matrix/Quaternion.hpp

```diff
@@ -280,9 +280,11 @@ class Quaternion : public Vector<Type, 4>
     void invert()
     {
         Quaternion &q = *this;
-        q(1) *= -1;
-        q(2) *= -1;
-        q(3) *= -1;
+        Type n = q.dot(q);
```

Comment:
Should have called the variable nsq to be clear. But for inverse believe it is conjugate over norm sq.